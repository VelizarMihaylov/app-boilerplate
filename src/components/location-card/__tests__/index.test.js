import React from 'react'
import { LocationCard } from '..'
import { create } from 'react-test-renderer'

import dayjs from 'dayjs'
jest.mock('dayjs')

describe('LocationCard', () => {
  dayjs.mockImplementation(() => ({
    extend: jest.fn(),
    utc: () => ({
      fromNow: () => '5 hours ago'
    })
  }))

  const measurements = [
    {
      parameter: 'no2',
      value: 37,
      lastUpdated: '2019-10-13T08:00:00.000Z',
      unit: 'µg/m³',
      sourceName: 'DEFRA',
      averagingPeriod: {
        value: 1,
        unit: 'hours'
      }
    },
    {
      parameter: 'so2',
      value: 3,
      lastUpdated: '2019-10-13T08:00:00.000Z',
      unit: 'µg/m³',
      sourceName: 'DEFRA',
      averagingPeriod: {
        value: 0.25,
        unit: 'hours'
      }
    },
    {
      parameter: 'pm25',
      value: 4,
      lastUpdated: '2019-10-13T08:00:00.000Z',
      unit: 'µg/m³',
      sourceName: 'DEFRA',
      averagingPeriod: {
        value: 24,
        unit: 'hours'
      }
    },
    {
      parameter: 'o3',
      value: 9,
      lastUpdated: '2019-10-13T08:00:00.000Z',
      unit: 'µg/m³',
      sourceName: 'DEFRA',
      averagingPeriod: {
        value: 8,
        unit: 'hours'
      }
    }
  ]
  it('should render with city, location and measurement props', () => {
    const LocationCardRender = create(
      <LocationCard
        city="Manchester"
        location="Manchester Piccadilly"
        measurements={measurements}
      />
    ).toJSON()
    expect(LocationCardRender).toMatchSnapshot()
  })
  it('should dispatch REMOVE_LOCATION action with the location prop set as a payload on onClick event', () => {
    const LocationCardRender = create(
      <LocationCard
        city="Manchester"
        location="Manchester Piccadilly"
        measurements={measurements}
        dispatch={jest.fn()}
      />
    ).root
    const CloseIcon = LocationCardRender.find(
      element =>
        element.type === 'svg' &&
        element.props.className === 'LocationCard--icon__close'
    )
    CloseIcon.props.onClick()
    expect(LocationCardRender.props.dispatch).toHaveBeenCalledWith({
      type: 'REMOVE_LOCATION',
      payload: 'Manchester Piccadilly'
    })
  })
})
