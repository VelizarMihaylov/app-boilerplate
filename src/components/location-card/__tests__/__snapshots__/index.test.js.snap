// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCard should render with city, location and measurement props 1`] = `
<div
  className="LocationCard"
>
  <svg
    className="LocationCard--icon__close"
    id="Capa_1"
    onClick={[Function]}
    style={
      Object {
        "enableBackground": "new 0 0 348.333 348.334",
      }
    }
    version="1.1"
    viewBox="0 0 348.333 348.334"
    x="0px"
    xlink="http://www.w3.org/1999/xlink"
    xmlSpace="preserve"
    xmlns="http://www.w3.org/2000/svg"
    y="0px"
  >
    <g>
      <path
        d="M336.559,68.611L231.016,174.165l105.543,105.549c15.699,15.705,15.699,41.145,0,56.85c-7.844,7.844-18.128,11.769-28.407,11.769c-10.296,0-20.581-3.919-28.419-11.769L174.167,231.003L68.609,336.563c-7.843,7.844-18.128,11.769-28.416,11.769c-10.285,0-20.563-3.919-28.413-11.769c-15.699-15.698-15.699-41.139,0-56.85l105.54-105.549L11.774,68.611c-15.699-15.699-15.699-41.145,0-56.844c15.696-15.687,41.127-15.687,56.829,0l105.563,105.554L279.721,11.767c15.705-15.687,41.139-15.687,56.832,0C352.258,27.466,352.258,52.912,336.559,68.611z"
      />
    </g>
  </svg>
  <div
    className="LocationCard--content"
  >
    <p
      className="LocationCard--content--updatedAt"
    >
      Updated 5 hours ago
    </p>
    <h3>
      Manchester Piccadilly
    </h3>
    <p>
      in Manchester, United Kingdom
    </p>
    <p
      className="LocationCard--content--values"
    >
      Value: NO2: 37, SO2: 3, PM25: 4, O3: 9
    </p>
  </div>
</div>
`;
